lectures/3-MatrixProductStates/InfiniteMPS.md

```diff
@@ -1,9 +1,347 @@
+:align: center
+```
+
+or alternatively we require ${\rm tr}(C^\dagger C) = 1$.
+
+
+### Expectation values revisited
+
+In the mixed gauge, we can locate the center site where the operator is acting, and then contract everything to the left and right to the identity to arrive at the particularly simple expression for the expectation value
+
+```{image} /_static/figures/imps/expVal3.svg
+:name: exp_val3
+:align: center
+```
+
+### Entanglement entropy
+
+The mixed canonical form with a diagonal $C$ now allows to straightforwardly write down a Schmidt decomposition of the state across an arbitrary bond in the chain
+
+```{math}
+\left | \Psi(A) \right \rangle = \sum_{i=1}^{D} C_i \left | \Psi^i_L(A_L) \right \rangle \otimes \left | \Psi^i_R(A_R) \right \rangle,
+```
+
+where the states $\left | \Psi^i_L(A_L) \right \rangle$ and $\left | \Psi^i_R(A_R) \right \rangle$ are orthogonal states on half the lattice. The diagonal elements $C_i$ are exactly the Schmidt numbers of any bipartition of the MPS, and as such determine its bipartite entanglement entropy
+
+```{math}
+S = -\sum_i C_i^2 \log(C_i^2) .
+```
+
+### Truncation
+
+The mixed canonical form also enables efficient truncatation of an MPS. The sum in the above Schmidt decomposition can be truncated, giving rise to a new MPS that has a reduced bond dimension for that bond. This truncation is optimal in the sense that the norm between the original and the truncated MPS is maximized. To arrive at a translation invariant truncated MPS, we can truncate the columns of the absorbed isometries $U$ and $V^\dagger$ correspondingly, thereby transforming *every* tensor $A_L$ or $A_R$. The truncated MPS in the mixed gauge is then given by
+
+```{image} /_static/figures/imps/truncMPS.svg
+:name: truncate_mps
+:align: center
+```
+
+We note that the resulting state based on this local truncation is not guaranteed to correspond to the MPS with a lower bond dimension that is globally optimal. This would require a variational optimization of the cost function.
+
+```{math}
+\left | \left | ~\left | \Psi(A) \right \rangle - \left | \Psi(\tilde{A}) \right \rangle ~\right | \right |^2.
+```
+
+
+### Code example: `MPSKit.InfiniteMPS`
+
+The Julia package [MPSKit.jl](https://github.com/maartenvd/MPSKit.jl) provides many tools for working with infinite MPS. Without going into much detail, we can already check some aspects of our discussion above with this numerical implementation.
+
+We can construct an [`MPSKit.InfiniteMPS`](https://maartenvd.github.io/MPSKit.jl/dev/lib/lib/#MPSKit.InfiniteMPS) by specifying the physical and virtual vector spaces of the MPS. We will use standard complex vector spaces as specified by a [`TensorKit.ComplexSpace`](https://jutho.github.io/TensorKit.jl/latest/lib/spaces/#TensorKit.ComplexSpace), and choose a physical dimension $d = 3$ and bond dimension $D = 5$.
+
+```{code-cell} julia
+using MPSKit, TensorKit
+
+d = 3 # physical dimension
+D = 5 # 
+mps = InfiniteMPS([ℂ^d], [ℂ^D])
```

Comment:
In principle this also works with integers, thus without the need of specifying TensorKit

Reply:
I did consider this but I thought it might be good to be more explicit, especially since we probably want people to get familiar with TensoKit spaces anyway. Do you think the integer constructor would actually be better.

As a side note, it's actually not documented in MPSKit that you can specify complex spaces using just integers I think.

Reply:
Hmm, I'm still rewriting most of these docs, I'll add it to that list.
It's also a bit flimsy at the moment, it works for some constructors but definitely not all, so it's probably best to get familiar with spaces